### PR TITLE
Cover LUKS encryption volumes with vol-resize

### PR DESCRIPTION
**Pull request by lento-sun**
Sign-offed by: Yi Sun(lentosun@163.com)

### CODE REVIEW

**Comment by lento-sun** [comment]
Test on x86_64 and found some libvirt issues. So  some of following cases failed because of libvirt bugs. Well, auto script should be ok.
Libvirt bugs:
https://bugzilla.redhat.com/show_bug.cgi?id=1371892
Another libvirt bug will be updated soon...

(01/12) type_specific.lento.virsh.vol_resize.positive_test.luks_encrypt.non_acl.sparse_capacity.dir_pool: FAIL (6.44 s)
 (02/12) type_specific.lento.virsh.vol_resize.positive_test.luks_encrypt.non_acl.sparse_capacity.fs_pool: FAIL (28.63 s)
 (03/12) type_specific.lento.virsh.vol_resize.positive_test.luks_encrypt.non_acl.sparse_capacity.netfs: FAIL (7.88 s)
 (04/12) type_specific.lento.virsh.vol_resize.positive_test.luks_encrypt.non_acl.allocate_capacity.dir_pool: FAIL (8.51 s)
 (05/12) type_specific.lento.virsh.vol_resize.positive_test.luks_encrypt.non_acl.allocate_capacity.fs_pool: FAIL (30.83 s)
 (06/12) type_specific.lento.virsh.vol_resize.positive_test.luks_encrypt.non_acl.allocate_capacity.netfs: FAIL (6.77 s)
 (07/12) type_specific.lento.virsh.vol_resize.positive_test.luks_encrypt.non_acl.delta_capacity.dir_pool: PASS (7.91 s)
 (08/12) type_specific.lento.virsh.vol_resize.positive_test.luks_encrypt.non_acl.delta_capacity.fs_pool: PASS (26.80 s)
 (09/12) type_specific.lento.virsh.vol_resize.positive_test.luks_encrypt.non_acl.delta_capacity.netfs: PASS (7.46 s)
 (10/12) type_specific.lento.virsh.vol_resize.positive_test.luks_encrypt.non_acl.delta_allocate_capacity.dir_pool: FAIL (8.48 s)
 (11/12) type_specific.lento.virsh.vol_resize.positive_test.luks_encrypt.non_acl.delta_allocate_capacity.fs_pool: FAIL (28.27 s)
 (12/12) type_specific.lento.virsh.vol_resize.positive_test.luks_encrypt.non_acl.delta_allocate_capacity.netfs: FAIL (7.51 s)

File: libvirt/tests/src/virsh_cmd/volume/virsh_vol_resize.py [hunk 1]
```diff
@@ -96,6 +100,73 @@ def get_expect_info(new_capacity, vol_path, resize_option=None):
...
+    sec_xml.usage = 'volume'
+    sec_xml.volume = vol_path
+    sec_xml.xmltreefile.write()
+
+    ret = virsh.secret_define(sec_xml.xml)
+    libvirt.check_exit_status(ret)
+    try:
+        encryption_uuid = re.findall(r".+\S+(\ +\S+)\ +.+\S+",
+                                     ret.stdout)[0].lstrip()
+    except IndexError, e:
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/volume/virsh_vol_resize.py]: `except IndexError:` should be fine if `e` is not required

File: libvirt/tests/src/virsh_cmd/volume/virsh_vol_resize.py [hunk 2]
```diff
@@ -96,6 +100,73 @@ def get_expect_info(new_capacity, vol_path, resize_option=None):
...
+    vol_arg['name'] = vol_name
+    volxml = libvirt_xml.VolXML()
+    newvol = volxml.new_vol(**vol_arg)
+    luks_encryption_params = {}
+    luks_encryption_params.update({"format": "luks"})
+    luks_encryption_params.update({"secret": {"type": "passphrase",
+                                              "uuid": sec_uuid}})
+    newvol.encryption = volxml.new_encryption(**luks_encryption_params)
+    vol_xml = newvol['xml']
+    if params.get('setup_libvirt_polkit') == 'yes':
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/volume/virsh_vol_resize.py]: you can include check for libvirt version and SKIP as in #1022 for acl related test case,
`libvirt_version.version_compare(1, 1, 1)`

File: libvirt/tests/src/virsh_cmd/volume/virsh_vol_resize.py (hunk 1 above)
Comment (will-Do) [inline libvirt/tests/src/virsh_cmd/volume/virsh_vol_resize.py]: yes, it is

File: libvirt/tests/src/virsh_cmd/volume/virsh_vol_resize.py (hunk 2 above)
Comment (lento-sun) [inline libvirt/tests/src/virsh_cmd/volume/virsh_vol_resize.py]: this is judged in run() line 250, we don't need to do it again in this method.

**Comment by lento-sun** [comment]
@will-Do changed according to above comments, and all autotest exceptions replaced by test.fail/cancel/error. pls have a check.

**Comment by dzhengfy** [comment]
@lento-sun please help double check this PR is both python3 and pytho2 compatible ASAP.  Thanks.

**Comment by lento-sun** [comment]
Done @dzhengfy @chunfuwen

**Pull request merged**